pkgs/applications/video/epgstation/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv, pkgs, makeWrapper, bash, nodejs }:
+
+let
+  nodePackages = import ./node-composition.nix {
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  };
+  workaround-opencollective-buildfailures = stdenv.mkDerivation {
+    # FIXME: This should be removed when a complete fix is available
+    # https://github.com/svanderburg/node2nix/issues/145
+    name = "workaround-opencollective-buildfailures";
+    phases = [ "installPhase" ];
+    installPhase = ''
+      mkdir -p $out/bin
+      touch $out/bin/opencollective-postinstall
+      chmod +x $out/bin/opencollective-postinstall
+    '';
+  };
+in
+nodePackages."epgstation-git+https://github.com/l3tnun/EPGStation.git#v1.7.4".override (drv: {
+  buildInputs = [ bash ];
+  nativeBuildInputs = [
+    workaround-opencollective-buildfailures
+    makeWrapper
+    nodePackages.node-pre-gyp
+  ];
+
+  preRebuild = ''
+    patch -p1 <<<"${builtins.readFile ./epgstation.patch}"
+  '';
+
+  postInstall = let
+    runtimeDeps = [ nodejs bash ];
+  in
+  ''
+    mkdir -p $out/{bin,libexec,share/doc/epgstation}
+
+    pushd $out/lib/node_modules/EPGStation
+
+    npm run build
```

Comment:
Should this really be done `postInstall`? (I have no npm experience to speak of)

Reply:
Yes, because I want this to run after `npm install`.

Reply:
To add more context, Epgstation isn't really a node package, and `npm install` won't build the application. The normal installation method is to `git clone`, run `npm install` to fetch dependencies, and then run `npm run build` to build the application.